model/log_iterator.go

```diff
@@ -327,3 +327,31 @@ func filterChunks(timeRange util.TimeRange, chunks []LogChunkInfo) []LogChunkInf
 
 	return filteredChunks
 }
+
+// LogIteratorHeap is a min-heap of LogIterator items.
+type LogIteratorHeap []LogIterator
+
+func (h LogIteratorHeap) Len() int { return len(h) }
+
+func (h LogIteratorHeap) Less(i, j int) bool {
+	return h[i].Item().Timestamp.Before(h[j].Item().Timestamp)
+}
+
+func (h LogIteratorHeap) Swap(i, j int) { h[i], h[j] = h[j], h[i] }
+
+func (h *LogIteratorHeap) Push(x interface{}) {
```

Comment:
I'd write type safe methods on this type so that you don't have to type coerce these interfaces (e.g. PushIterator and PopIterator or just Add/Next) in the calling code.

Reply:
added below